pkg/account/component.go

```diff
@@ -541,3 +543,51 @@ func (c *component) executeActions(ctx context.Context, actions []string) error
 
 	return err
 }
+
+func (c *component) CancelEmailChange(ctx context.Context) error {
+	accessToken := ctx.Value(cs.CtContextAccessToken).(string)
+	realm := ctx.Value(cs.CtContextRealm).(string)
+
+	// Get the "old" user representation from Keycloak
+	oldUserKc, err := c.keycloakAccountClient.GetAccount(accessToken, realm)
+	if err != nil {
+		c.logger.Warn(ctx, "msg", "Can't get account", "err", err.Error())
+		return err
+	}
+	keycloakb.ConvertLegacyAttribute(&oldUserKc)
+
+	oldUserKc.SetAttributeString(constants.AttrbEmailToValidate, "")
```

Comment:
What if emailToValidate is already missing/blank? Should we update such an account?